### DISCUSSION TITLE
How do you use the <FCCExcludeFromCodeCoverage/> csproj attribute?

### DISCUSSION THREAD

**Discussion by fmcgarry** [FortuneN/FineCodeCoverage#186] [Q&A]
I've tried to put in in the csproj file for a project that I did not want included in the coverage, but after running the analysis, the project still shows up in the coverage window. Am I misunderstanding how the <FCCExcludeFromCodeCoverage/> attribute is supposed to be used?

**Comment by tonyhallett** [comment]
Please see the zipped solution.

[FCCExcludeFromCodeCoverage.zip](https://github.com/FortuneN/FineCodeCoverage/files/7184733/Delete_FCCExcludeFromCodeCoverage.zip)

If you are still having trouble then can you provide the text in the FCC output window pane.
![image](https://user-images.githubusercontent.com/11292998/133766058-956904be-d70e-42da-b025-6e1e9e7fa843.png)

**Reply by fmcgarry** [reply]
The solution you provided worked out of the box. However, for some reason, it's still not picking up the attribute in my projects. The following are differences between our solutions and what I've tried:

- I'm using .NET Core 3.1 instead of .NET 5. I downgraded and it did not change anything.
- I'm using MSTest instead of NUnit. Changing it didn't affect anything.
- I was missing the `--exclude "[FCCExcludeFromCodeCoverage]*"` from mine, but adding it in didn't help.

![image](https://user-images.githubusercontent.com/25107011/133824708-866e0b8f-1316-4a13-ad6e-9773a853c97f.png)

Is there a way to reset the options back to default? Maybe I somehow messed them up since I was missing `--exclude "[FCCExcludeFromCodeCoverage]*"`.

**Reply by tonyhallett** [reply]
The exclude "[FCCExcludeFromCodeCoverage]*" is the result of applying the attribute to a project called FCCExcludeFromCodeCoverage that is referenced by a test project.

You will have to provide your solution for me to comment further. In particular the .csproj file that you have annotated.

**Reply by tonyhallett** [reply]
Also framework version and test framework should be irrelevant.

**Reply by tonyhallett** [reply]
Can you show how you are referencing assemblies from the test project ?

You could just use the assembly attribute ExcludeFromCodeCoverage.  FCCExcludeFromCodeCoverage just is a helper that provides the same functionality for .NET Framework.
